src/ui/ui.ts

```diff
@@ -211,10 +383,15 @@ export class UI extends Phaser.GameObjects.Container {
   public resetHandlers(): void {
     this.mode = DEFAULT_MODE;
     const currentHandler = this.getCurrentHandler();
-    for (const handler of this.handlers.filter((h) => h.active && h !== currentHandler)) {
-      handler.stop();
+    this.handlers.forEach((handler) => {
+      if (handler.active && handler !== currentHandler) {
+        handler.stop();
+      }
+    });
```

Comment:
I'd prefer to keep it this way (so that `handler` is defined as `const`) like it was originally.
```suggestion
    for (const handler of this.handlers) {
      if (handler.active && handler !== currentHandler) {
        handler.stop();
      }
    }
```

Reply:
no clue why I did it like this 😂